mesonbuild/backend/ninjabackend.py

```diff
@@ -421,6 +422,7 @@ def generate_target(self, target, outfile):
             obj_list.append(self.generate_single_compile(target, outfile, src, 'vala', [], header_deps))
 
         # Generate compile targets for all the pre-existing sources for this target
+        preexisting_header_deps = header_deps + [os.path.join('.', header) for header in vala_generated_headers]
```

Comment:
My test-case breaks without it because the library is in the root, so each item is "naked" without any directory components, i.e. just the filename, and this results in the generated .ninja pointing at the wrong location (inside the `@sta` directory, instead of above it).